File: web/src/app/service/admin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Admin} from "../entity/admin";
import {ResponseBody} from "../entity/response-body";

@Injectable({
  providedIn: 'root'
})
export class AdminService {
  private baseUrl = 'http://localhost:8080/Admin';

  constructor(private httpClient: HttpClient) { }

  add(admin: {name: string, username: string, ano: string, role: number}): Observable<ResponseBody> {
    return this.httpClient.post<ResponseBody>(`${this.baseUrl}/add`, admin);
  }

  getAll(): Observable<Admin[]> {
    return this.httpClient.get<Admin[]>(`${this.baseUrl}/getAll`);
  }

  delete(id: number): Observable<ResponseBody> {
    return this.httpClient.delete<ResponseBody>(`${this.baseUrl}/delete/${id}`);
  }

  resetPassword(id: number, newPassword: string): Observable<ResponseBody> {
    return this.httpClient.patch<ResponseBody>(`${this.baseUrl}/resetPassword/${id}`, newPassword);
  }

  searchAdmins(name: string, ano: string): Observable<Admin[]> {
    const params = { name, ano };
    return this.httpClient.get<Admin[]>(`${this.baseUrl}/search`, { params });
  }

  getAdminById(id: number): Observable<Admin> {
    return this.httpClient.get<Admin>(`${this.baseUrl}/getAdminById/${id}`);
  }

  updateAdmin(admin: any): Observable<ResponseBody> {
    return this.httpClient.patch<ResponseBody>(`${this.baseUrl}/update/${admin.id}`, admin);
  }
}

File: api/src/main/java/com/mengyunzhi/eduPlanner/service/AdminService.java
package com.mengyunzhi.eduPlanner.service;

import com.mengyunzhi.eduPlanner.dto.AdminRequest;
import com.mengyunzhi.eduPlanner.dto.Response;
import com.mengyunzhi.eduPlanner.entity.Admin;

import java.util.List;

public interface AdminService {
    /**
     * 新增管理员
     * @param adminRequest
     * @return
     */
    Response<Void> save(AdminRequest adminRequest);

    /**
     * 获取所有管理员
     * @return
     */
    List<Admin> getAll();

    Response<Void> delete(Long id);

    Response<Admin> resetPassword(Long id, String newPassword);

    List<Admin> searchAdmins(String name, String ano);

    Admin getAdminById(Long id);

    Response<Admin> updateAdmin(Long id, AdminRequest adminRequest);
}
